src/components/ContentWrapper.jsx

```diff
@@ -68,30 +68,31 @@ const ContentWrapper = (props) => {
   const gem2sRunningError = backendErrors.includes(gem2sStatusKey);
   const completedGem2sSteps = backendStatus.gem2s?.completedSteps;
 
-  const changedQCFilters = useSelector((state) => state.experimentSettings.processing.meta.changedQCFilters);
+  const changedQCFilters = useSelector(
+    (state) => state.experimentSettings.processing.meta.changedQCFilters,
+  );
 
   // This is used to prevent a race condition where the page would start loading immediately
   // when the backend status was previously loaded. In that case, `backendLoading` is `false`
   // and would be set to true only in the `loadBackendStatus` action, the time between the
   // two events would allow pages to load.
   const [backendStatusRequested, setBackendStatusRequested] = useState(false);
-
   const [changesNotAppliedModalPath, setChangesNotAppliedModalPath] = useState(null);
 
-  const setUpConnection = async () => {
-    const io = await connectionPromise;
-    const cb = experimentUpdatesHandler(dispatch);
-    io.on(`ExperimentUpdates-${experimentId}`, (update) => cb(experimentId, update));
-  };
-
   useEffect(() => {
     if (!experimentId) {
       return;
     }
 
     dispatch(loadBackendStatus(experimentId));
 
-    setUpConnection();
+    (async () => {
```

Comment:
this is subjective, but it seemed neater to me when it was written as a `setUpConnection` function. Are there any particular reasons to write it as an anonymous function?

Reply:
Not necessarily, but I feel like it being refactored to a different place is misleading. We are essentially creating a promise by calling an `async` function, but we are not waiting for this promise to be resolved. This is a potentially dangerous operation, and putting the async function inline means people would be less likely to assume that `setUpConnection` is not an async function.

Reply:
fair enough, happy to have it like this